llama-index-integrations/vector_stores/llama-index-vector-stores-faiss/llama_index/vector_stores/faiss/map_store.py

```
@@ -0,0 +1,188 @@
+                "Please create an IndexIDMap2 index and pass it to the FaissVectorMapStore."
+            )
+        super().__init__(faiss_index=faiss_index)
+        self._node_id_to_faiss_id_map = {}
+        self._faiss_id_to_node_id_map = {}
+
+    def add(
+        self,
+        nodes: List[BaseNode],
+        **add_kwargs: Any,
+    ) -> List[str]:
+        """
+        Add nodes to index.
+
+        NOTE: in the Faiss vector store, we do not store text in Faiss.
+
+        Args:
+            nodes: List[BaseNode]: list of nodes with embeddings
+
+        """
+        new_ids = []
+        for node in nodes:
+            text_embedding = node.get_embedding()
+            text_embedding_np = np.array(text_embedding, dtype="float32")[np.newaxis, :]
+            self._node_id_to_faiss_id_map[node.id_] = self._faiss_index.ntotal
+            self._faiss_id_to_node_id_map[self._faiss_index.ntotal] = node.id_
+            self._faiss_index.add_with_ids(text_embedding_np, self._faiss_index.ntotal)
+            new_ids.append(node.id_)
+        return new_ids
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes using with ref_doc_id.
+
+        Args:
+            ref_doc_id (str): The doc_id of the document to delete.
+
+        """
+        # only handle delete on node_ids
+        if ref_doc_id in self._node_id_to_faiss_id_map:
+            faiss_id = self._node_id_to_faiss_id_map[ref_doc_id]
+            # remove the faiss id from the faiss index
+            self._faiss_index.remove_ids(np.array([faiss_id], dtype=np.int64))
+            # remove the node id from the node id map
+            del self._node_id_to_faiss_id_map[ref_doc_id]
+            # remove the faiss id from the faiss id map
+            del self._faiss_id_to_node_id_map[faiss_id]
+
+    def delete_nodes(
+        self,
+        node_ids: Optional[List[str]] = None,
+        **delete_kwargs: Any,
```

technically the `delete_nodes` method accepts node_ids and filters (we can't delete by filters, so we should raise an error)